src/box/memtx_tx.c

```diff
@@ -2943,6 +2944,17 @@ memtx_tx_on_space_delete(struct space *space)
 		if (story->del_stmt != NULL)
 			memtx_tx_history_remove_stmt(story->del_stmt);
 		memtx_tx_story_full_unlink_on_space_delete(story);
+		for (uint32_t i = 0; i < story->index_count; i++) {
+			while (!rlist_empty(&story->link[i].read_gaps)) {
+				struct rlist *read_gaps =
+					&story->link[i].read_gaps;
+				struct gap_item_base *item =
+					rlist_first_entry(read_gaps,
+							  struct gap_item_base,
+							  in_read_gaps);
+				memtx_tx_delete_gap(item);
+			}
```

Comment:
Nit: it would be more convenient to write `struct rlist *read_gaps = &story->link[i].read_gaps;` before the `while` loop.

Reply:
Fixed.